milli/src/update/index_documents/transform.rs

```diff
@@ -677,6 +774,39 @@ impl<'a, 'i> Transform<'a, 'i> {
     }
 }
 
+/// Merge all the obks in the order we see them.
+fn merge_obkvs_and_operations<'a>(_key: &[u8], obkvs: &[Cow<'a, [u8]>]) -> Result<Cow<'a, [u8]>> {
+    // [add, add, delete, add, add]
+    // we can ignore everything that happened before the last delete.
+    let starting_position = obkvs
+        .iter()
+        .rev()
+        .position(|obkv| obkv[0] == Operation::Deletion as u8)
+        .map_or(0, |pos| obkvs.len() - pos);
+
+    // [add, add, delete]
+    // if the last operation was a deletion then we simply return the deletion
+    if starting_position == obkvs.len() {
+        return Ok(obkvs[obkvs.len() - 1].clone());
+    }
```

Comment:
Oh yes, true, my code was correct, but it isn't after the use of position.

I still took your code because I find it easier to read, and fixed my condition + added a test; see https://github.com/meilisearch/meilisearch/pull/3470/commits/fb5e4957a6f0359cc045e02c826ad2ba88b6f2ac